lib/cloud/clients.go

```diff
@@ -284,26 +340,107 @@ func (c *cloudClients) initAzureCredential() (azcore.TokenCredential, error) {
 		return c.azureCredential, nil
 	}
 	logrus.Debug("Initializing Azure default credential chain.")
-	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	// TODO(gavin): if/when we support AzureChina/AzureGovernment, we will need to specify the cloud in these options
+	options := &azidentity.DefaultAzureCredentialOptions{}
+	cred, err := azidentity.NewDefaultAzureCredential(options)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	c.azureCredential = cred
 	return cred, nil
 }
 
+func (c *cloudClients) initAzureMySQLClient(subscription string) (azure.DBServersClient, error) {
+	cred, err := c.GetAzureCredential()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
Credential is cached so it is not created each time, but the reason I did it above the subscription check was to avoid deadlock. I could weave multiple lock/unlock, but I don't think we're gaining much performance-wise